kuwala/core/backend/app/controller/data_source/snowflake.py

```diff
@@ -0,0 +1,181 @@
+import functools
+
+from database.schemas.data_source import ConnectionParameters
+from fastapi import HTTPException
+import snowflake.connector
+
+
+def map_connection_parameters(connection_parameters: ConnectionParameters):
+    user = connection_parameters.user
+    password = connection_parameters.password
+    account = connection_parameters.organization + "-" +  connection_parameters.account
+    database = connection_parameters.database
+    warehouse = connection_parameters.warehouse
+    role = connection_parameters.role
+
+    if not (user and password and account and database):
```

Comment:
Aren't `warehouse` and `role` also required?

Reply:
Yes they are, I just added them to the latest commit.